vsm

```diff
@@ -147,31 +149,55 @@ def process(state, signal, value):
     state.got_signal(signal, value)
 
 
-def run(state):
+def receive():
     '''
-        Loop to grab logic from stdin, then spawn a threads for processing
+        Loop to grab logic from stdin
     '''
     for line in sys.stdin:
         line = line.replace(" ", "").strip('\n')
         if line == "":
             # Ignore blank lines
             pass
         if line == "quit":
-            break
+            exit(0)
         elif "=" in line:
             line = line.split("=")
-            process(state, line[0], line[1])
+            return (line[0], line[1])
         else:
-            process(state, line, None)
-
+            return (line, None)
+
+def run(state, no_loop):
+    try:
+        while True:
+            message = receive()
+            if message == None:
+                exit(0)
+            signal, value = message
+            process(state, signal, value)
+            if no_loop:
```

Comment:
Ok, indeed, this is a bit problematic for tests sending more than one message (and I had to skip those for the IPC module at the end).

I also thought about your two suggestions, and I tried killing the process but that (for some reason I still don't figure out) didn't work, and using a command to exit worked just fine, so I guess I will simply go that way.